sources/engine/Xenko.VirtualReality/OpenVR/OpenVR.cs

```diff
@@ -150,6 +151,34 @@ public void Update()
             }
         }
 
+        public class TrackedDevice
+        {
+            public TrackedDevice(int trackerIndex)
+            {
+                TrackerIndex = trackerIndex;
+            }
+
+            const int StringBuilderSize = 64;
+            StringBuilder serialNumberStringBuilder = new StringBuilder(StringBuilderSize);
+            public void Update()
+            {
+                var error = ETrackedPropertyError.TrackedProp_Success;
+                serialNumberStringBuilder.Clear();
+                Valve.VR.OpenVR.System.GetStringTrackedDeviceProperty((uint)TrackerIndex, ETrackedDeviceProperty.Prop_SerialNumber_String, serialNumberStringBuilder, StringBuilderSize, ref error);
+                if (error == ETrackedPropertyError.TrackedProp_Success)
+                    SerialNumber = serialNumberStringBuilder.ToString();
```

Comment:
Does that mean it will allocate every frame? (something we are trying to avoid)

Reply:
what do you mean by "allocate every frame"? what those lines do, is accessing the serialnumber property of the trackeddevice with the index TrackerIndex. i don't see anything getting allocated there.

Reply:
`StringBuilder.ToString()` will allocate a new string at every update.

Reply:
ok, any idea to improve that?